### PR TITLE
Port tests from phoenix.gen.{html,json,model} to phx.gen.{html,json,schema}

### PR DESCRIPTION
**Pull request by gavlak**
Ref: #2131 
I ported almost all tests from phoenix.gen.{html,json,model} to phx.gen.{html,json,schema}. There are only a few failing tests. Other ported tests are working just fine.

### CODE REVIEW

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: test/mix/tasks/phx.gen.schema_test.exs [hunk 1]
```diff
@@ -36,6 +36,98 @@ defmodule Mix.Tasks.Phx.Gen.SchemaTest do
...
+  end
+
+  test "generates belongs_to associations with association table provided by user", config do
+    in_tmp_project config.test, fn ->
+      Gen.Schema.run(~w(Blog.Post posts title user_id:references:users))
+
+      assert [migration] = Path.wildcard("priv/repo/migrations/*_create_blog_post.exs")
+
+      assert_file migration, fn file ->
+        assert file =~ "add :user_id, {:references, :users}"
```
Comment (josevalim) [inline test/mix/tasks/phx.gen.schema_test.exs]: This is incorrect :) It should be a belongs_to. :)

Comment (gavlak) [inline test/mix/tasks/phx.gen.schema_test.exs]: My bad, I'll fix it. But, shouldn't the belongs_to be in schema instead of migration?

**Comment by josevalim** [comment]
No worries, I just wanted to point it out. It is already fixed in master :)
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by gavlak** [comment]
Just had a look at master. Looking good! 👍

**Pull request merged**